File: java/src/main/java/club/marzipan/javacarrentals/Vehicle.java
package club.marzipan.javacarrentals;

import java.util.UUID;

public class Vehicle {

    public enum VehicleClass {
        standard_small, standard_large, premium_small, premium_large
    }

    private UUID id;
    private String licensePlate;
    private String vin;
    private String year;
    private String make;
    private String model;
    private VehicleClass vehicleClass;

    public UUID getId() {
        return this.id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public String getLicensePlate() {
        return this.licensePlate;
    }

    public void setLicensePlate(String licensePlate) {
        this.licensePlate = licensePlate;
    }

    public String getVin() {
        return this.vin;
    }

    public void setVin(String vin) {
        this.vin = vin;
    }

    public String getYear() {
        return this.year;
    }

    public void setYear(String year) {
        this.year = year;
    }

    public String getMake() {
        return this.make;
    }

    public void setMake(String make) {
        this.make = make;
    }

    public String getModel() {
        return this.model;
    }

    public void setModel(String model) {
        this.model = model;
    }

    public VehicleClass getVehicleClass() {
        return this.vehicleClass;
    }

    public void setVehicleClass(VehicleClass vehicleClass) {
        this.vehicleClass = vehicleClass;
    }

}

File: rust/src/vehicle.rs
use derive_more::{Display, From, Into};
use serde::Serialize;

#[derive(Debug, PartialEq, Eq, Clone, Hash, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct Vehicle {
    pub id: Id,
    pub license_plate: LicensePlate,
    pub vin: Vin,
    pub year: u32,
    pub make: Make,
    pub model: Model,
    pub vehicle_class: VehicleClass,
}

#[derive(Debug, PartialEq, Eq, Copy, Clone, Hash, Serialize, From, Into, Display)]
pub struct Id(pub uuid::Uuid);

#[derive(Debug, PartialEq, Eq, Clone, Hash, Serialize, From, Into, Display)]
pub struct LicensePlate(pub String);

#[derive(Debug, PartialEq, Eq, Clone, Hash, Serialize, From, Into, Display)]
pub struct Vin(pub String);

#[derive(Debug, PartialEq, Eq, Clone, Hash, Serialize, From, Into, Display)]
pub struct Make(pub String);

#[derive(Debug, PartialEq, Eq, Clone, Hash, Serialize, From, Into, Display)]
pub struct Model(pub String);

#[derive(Debug, PartialEq, Eq, Clone, Hash, Serialize, From, Display)]
#[serde(rename_all = "lowercase")]
pub enum VehicleClass {
    StandardSmall,
    StandardLarge,
    PremiumSmall,
    PremiumLarge,
}
